components/director/internal/domain/webhook/datainputbuilder/cert_subject_builder.go

```diff
@@ -0,0 +1,71 @@
+package datainputbuilder
+
+import (
+	"context"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/pkg/webhook"
+	"github.com/pkg/errors"
+)
+
+//go:generate mockery --exported --name=certSubjectRepository --output=automock --outpkg=automock --case=underscore --disable-version-string
+type certSubjectRepository interface {
+	ListByConsumerID(ctx context.Context, consumerID string) ([]*model.CertSubjectMapping, error)
+}
+
+// WebhookCertSubjectBuilder takes care to get and build trust details for objects
+type WebhookCertSubjectBuilder struct {
+	certSubjectRepository certSubjectRepository
+}
+
+// NewWebhookCertSubjectBuilder creates a WebhookCertSubjectBuilder
+func NewWebhookCertSubjectBuilder(certSubjectRepository certSubjectRepository) *WebhookCertSubjectBuilder {
+	return &WebhookCertSubjectBuilder{
+		certSubjectRepository: certSubjectRepository,
+	}
+}
+
+// GetTrustDetailsForObjects builds trust details for objects with IDs objectIDs
+func (b *WebhookCertSubjectBuilder) GetTrustDetailsForObjects(ctx context.Context, objectIDs []string) (map[string]*webhook.TrustDetails, error) {
+	if len(objectIDs) == 0 {
+		return nil, nil
+	}
+
+	result := make(map[string]*webhook.TrustDetails, len(objectIDs))
+
+	for _, objectID := range objectIDs {
+		certSubjectMappingsForObject, err := b.certSubjectRepository.ListByConsumerID(ctx, objectID)
+		if err != nil {
+			return nil, errors.Wrapf(err, "while listing cert subject mappings for consumer with ID %q", objectID)
+		}
+
+		result[objectID] = &webhook.TrustDetails{Subjects: certSubjectMappingsToCertificateDetails(certSubjectMappingsForObject)}
+	}
+
+	return result, nil
+}
+
+// GetTrustDetailsForObject builds trust details for objects with ID objectID
+func (b *WebhookCertSubjectBuilder) GetTrustDetailsForObject(ctx context.Context, objectID string) (*webhook.TrustDetails, error) {
+	trustDetails, err := b.GetTrustDetailsForObjects(ctx, []string{objectID})
+	if err != nil {
+		return nil, err
+	}
+
+	result, ok := trustDetails[objectID]
+	if !ok {
+		return nil, errors.Errorf("There are no trust details for object with ID %q", objectID)
```

Comment:
Are there cases where there won't be any trust details and if we don't have any, should we throw an error?

Reply:
Currently, I think it is impossible to enter that `if` case because even if there are no trust details for the object the `GetTrustDetailsForObjects` func will return a map where there is an entry with key `objectID` and empty value(`webhook.TrustDetails`). So it's practically impossible to enter that case but I wrote it because it's a bad pattern not to check if there is an entry with a given key in a map.